undertow/src/main/java/org/wildfly/extension/undertow/filters/ModClusterDefinition.java

```diff
@@ -325,26 +335,72 @@ public HttpHandler createHttpHandler(Predicate predicate, ModelNode model, HttpH
 
     }
 
+    @Override
+    public void registerOperations(ManagementResourceRegistration parent) {
+        ResourceDescriptor descriptor = new ResourceDescriptor(this.getResourceDescriptionResolver())
+                .addCapabilities(Capability.class)
+                .addRequiredSingletonChildren(SingleAffinityResourceDefinition.PATH)
+                ;
+
+        for (AttributeDefinition attribute : ATTRIBUTES) {
+            descriptor.addAttributes(new Attribute() {
+                @Override
+                public AttributeDefinition getDefinition() {
+                    return attribute;
+                }
+            });
+        }
+
+        ModClusterResourceServiceHandler handler = new ModClusterResourceServiceHandler();
```

Comment:
I don't think you need a custom ResourceServiceHandler implementation.  Just use ParentResourceServiceHandler.
This will allow you to leverage the RestartParentResource*Handlers.

Reply:
Nevermind - this isn't easily done without significant refactoring, since ModClusterService doesn't properly encapsulate service installation.